src/JsonApiQueryParser.js

```diff
@@ -225,6 +236,39 @@ class JsonApiQueryParser {
   }
 
   /**
+   * [Note: The are no proper specifications for this parameter yet.
+   * For now the filter is implemented similar to the fields parameter. Values should be url encoded to allow for special characters.]
+   *
+   * @param {[string]} filterString [Required sort query string piece. Example: "filter[name][like]=John%20Doe".]
+   * @param {[object]} requestDataSubset [Required reference to the requestData.queryData object.]
+   * @return {[object]} requestDataSubset [Returning the modified request data.]
+   *
+   **/
+  static parseFilterType (filterString, requestDataSubset) {
+    let targetType;
+    let targetColumn;
+    let targetFilterString;
+    
+    targetType = filterString.replace(PARSE_PARAM.parseFilterType, function(match, $1) {
+      return $1;
+    });
+    
+    targetColumn = filterString.replace(PARSE_PARAM.parseFilterType, function(match, $1, $2) {
+      return $2;
+    });
+
+    targetFilterString = filterString.replace(PARSE_PARAM.parseFilterType, function(match, $1, $2, $3) {
+      return $3;
+    });
+
+    if(requestDataSubset.filter[targetType]){
```

Comment:
The targetType and targetColumn seem to be mixed up here, it will try set filter['name'] instead of using the type (not,lt,gt etc)
string 'filter[name][not]=jack' attempts filter.name.not = 'jack'   --- Should be filter.not.name = jack
string 'filter[age][lt]=24' attempts filter.age.lt = 24  --- Should be filter.lt.age = 24